bfillings/vsearch.py

```diff
@@ -0,0 +1,580 @@
+        """
+        # The list of values which can be passed on a per-run basis
+        allowed_values = ['--uc', '--output', '--sortbysize',
+                          '--consout', '--uchime_denovo',
+                          '--derep_fulllength', '--maxuniquesize',
+                          '--minuniquesize', '--sizein',
+                          '--sizeout', '--strand', '--threads',
+                          '--uchime_ref', '--chimeras',
+                          '--nonchimeras', '--db', '--uchimeout',
+                          '--blast6out', '--abskew',
+                          '--sortbysize', '--maxsize', '--minsize']
+
+        unsupported_parameters = set(data.keys()) - set(allowed_values)
+        if unsupported_parameters:
+            raise ApplicationError(
+                "Unsupported parameter(s) passed when calling vsearch: %s" %
+                ' '.join(unsupported_parameters))
+
+        for v in allowed_values:
+            # turn the parameter off so subsequent runs are not
+            # affected by parameter settings from previous runs
+            self.Parameters[v].off()
+            if v in data:
+                # turn the parameter on if specified by the user
+                self.Parameters[v].on(data[v])
+
+        return ''
+
+    def _get_result_paths(self, data):
+        """ Set the result paths """
+
+        result = {}
+
+        result['Output'] = ResultPath(
+            Path=self.Parameters['--output'].Value,
+            IsWritten=self.Parameters['--output'].isOn())
+
+        result['ClusterFile'] = ResultPath(
+            Path=self.Parameters['--uc'].Value,
+            IsWritten=self.Parameters['--uc'].isOn())
+
+        # uchime 3-way global alignments
+        result['Output_aln'] = ResultPath(
+            Path=self.Parameters['--uchimealns'].Value,
+            IsWritten=self.Parameters['--uchimealns'].isOn())
+
+        # uchime tab-separated format
+        result['Output_tabular'] = ResultPath(
+            Path=self.Parameters['--uchimeout'].Value,
+            IsWritten=self.Parameters['--uchimeout'].isOn())
+
+        # chimeras fasta file output
+        result['Output_chimeras'] = ResultPath(
+            Path=self.Parameters['--chimeras'].Value,
+            IsWritten=self.Parameters['--chimeras'].isOn())
+
+        # nonchimeras fasta file output
+        result['Output_nonchimeras'] = ResultPath(
+            Path=self.Parameters['--nonchimeras'].Value,
+            IsWritten=self.Parameters['--nonchimeras'].isOn())
+
+        # log file
+        result['LogFile'] = ResultPath(
+            Path=self.Parameters['--log'].Value,
+            IsWritten=self.Parameters['--log'].isOn())
+
+        return result
+
+    def getHelp(self):
+        """Method that points to documentation"""
+        help_str = """
+        VSEARCH is hosted at:
+        https://github.com/torognes/vsearch
+
+        The following papers should be cited if this resource is used:
```

Comment:
I would delete this line and the next and instead say: *Please cite the above URL if this wrapper is used in published work.*